### DISCUSSION TITLE
why in release 2.8, there is no go.mod

### DISCUSSION THREAD

**Discussion by nanyi545** [distribution/distribution#4317] [General]
I am pretty new to go development...   why in release 2.8, there is no go.mod, i do not quite get it...

**Comment by milosgajdos** [comment]
`v2.x` branches had been created during the distribution Jurassic era during which Go modules did not exist.

When we restarted the work on distribution we decided we'd maintain `v2.x` release branches only for security updates and very very minor patches and focus on `v3` release. We've also decided that the `main` branch will become the release branch as opposed to maintaining release branches for each release.

Now, Go modules require a very specific handling of v1+ versions which we figured would be rather painful to deal with on the old release branch, so we invested all the effort into Gomodding the `main` branch that has become the new release branch.

I'd strongly recommend using the `main` branch and `v3` release in every possible way as it contains a tremendous number of bug fixes, performance improvements etc. There will not be any more updates to `v2` and it will essentially become deprecated this year.

**Reply by nanyi545** [reply]
thank you very much for this explanation
